src/migrator/sync.ts

```diff
@@ -0,0 +1,177 @@
+import {
+	modelText,
+	MigrationTuple,
+	MigrationError,
+	defaultMigrationCategory,
+	checkModelAlreadyExists,
+	setExecutedMigrations,
+	getExecutedMigrations,
+	lockMigrations,
+	MigrationCategories,
+	RunnableMigrations,
+} from './utils';
+import type { Tx } from '../database-layer/db';
+import type { Config, Model } from '../config-loader/config-loader';
+
+import * as _ from 'lodash';
+import * as sbvrUtils from '../sbvr-api/sbvr-utils';
+
+type ApiRootModel = Model & { apiRoot: string };
+
+export const postRun = async (tx: Tx, model: ApiRootModel): Promise<void> => {
+	const { initSql } = model;
+	if (initSql == null) {
+		return;
+	}
+
+	const modelName = model.apiRoot;
+
+	const exists = await checkModelAlreadyExists(tx, modelName);
+	if (!exists) {
+		(sbvrUtils.api.migrations?.logger.info ?? console.info)(
+			'First time executing, running init script',
+		);
+
+		await lockMigrations(tx, modelName, async () => {
+			try {
+				await tx.executeSql(initSql);
+			} catch (err) {
+				(sbvrUtils.api.migrations?.logger.error ?? console.error)(
+					`postRun locked sql execution error ${err} `,
+				);
+				throw new MigrationError(err);
+			}
+		});
+	}
+};
+
+export const run = async (tx: Tx, model: ApiRootModel): Promise<void> => {
+	const { migrations } = model;
+	if (
+		migrations == null ||
+		_.isEmpty(migrations) ||
+		migrations[defaultMigrationCategory] === undefined
+	) {
+		return;
+	}
+
+	let runMigrations: RunnableMigrations = {};
+	for (const [key, value] of Object.entries(migrations)) {
+		if (key === defaultMigrationCategory && value instanceof Object) {
+			runMigrations = { ...runMigrations, ...value };
+		} else if (!(key in MigrationCategories)) {
+			runMigrations = { ...runMigrations, ...{ [key]: value } };
+		}
+	}
```

Comment:
I think you can simplify this to
```suggestion
	const runMigrations: RunnableMigrations = defaultMigrationCategory in migrations ? migrations[defaultMigrationCategory] : migrations;
```
since you either have specify using the migrations categories or not specify at all - we don't allow mixing the 2 versions in the types and I don't think it makes sense for the code to allow it either (if you want you can also explicitly check that there's no non-migration category entries when using the migration category form)

Reply:
Need to add a check if migrations[defaultMigrationCategory] is object and not string or function. It's a double ternary, which is 'still readable'